server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyServerHandler.scala

```diff
@@ -20,11 +20,18 @@ import sttp.tapir.server.netty.NettyResponseContent.{
 import sttp.tapir.server.netty.{NettyResponse, NettyServerRequest, Route}
 
 import scala.collection.JavaConverters._
+import scala.collection.mutable.{Queue => MutableQueue}
+import scala.concurrent.Future
 
-class NettyServerHandler[F[_]](route: Route[F], unsafeRunAsync: (() => F[Unit]) => Unit, maxContentLength: Option[Int])(implicit
-    me: MonadError[F]
+class NettyServerHandler[F[_]](route: Route[F], unsafeRunAsync: (() => F[Unit]) => (() => Future[Unit]), maxContentLength: Option[Int])(
```

Comment:
I'd add a comment that `unsafeRunAsync` returns a function to cancel the execution

Reply:
Comment added, and signature extended, it's even more complex now ;)